src/main/java/uk/gov/hmcts/reform/idam/ApplicationExecutor.java

```diff
@@ -18,12 +19,14 @@ public class ApplicationExecutor implements ApplicationRunner {
 
     private final IdamUserDisposerService service;
 
+    private final SecurityUtil securityUtil;
+
     @Override
     @SuppressWarnings("PMD.AvoidThrowingRawExceptionTypes")
     public void run(ApplicationArguments args) {
-
         if (isServiceEnabled) {
             log.info("Starting the Idam-Disposer job...");
+            securityUtil.generateTokens();
```

Comment:
nothing stopping you to generate the tokens before the _while_ loop in the run() method.

Reply:
There is no while loop and if service is not enabled then why we want to generate the token

Reply:
I meant in IdamUserDisposerService.run() method.